src/customHooks/useValidateInput/index.js

```diff
@@ -0,0 +1,96 @@
+import { useState } from "react";
+
+const useValidateInput = (
+  initVal,
+  emptyValidation = { emptyErrMsg: "값을 입력해주세요" },
+  regValidation = {
+    regExp: /$/,
+    regExpErrMsg: "옳바른 값을 입력해주세요"
+  },
+  apiValidation = {
+    apiCall: () => new Promise(res => res({ success: true })),
+    apiCallPendingMsg: "잠시만 기다려주세요",
+    apiCallErrMsg: "사용할 수 없는 값입니다"
+  },
+  matchValidation = {
+    matchVal: "",
+    matchValErrMsg: "값이 일치하지 않습니다"
+  }
```

Comment:
```suggestion
const VALIDATAION_INITIAL_VALES = {
empty: {emptyErrMsg: "값을 입력해주세요"},
reg: {
    regExp: /$/,
    regExpErrMsg: "옳바른 값을 입력해주세요"
},
api : {
    apiCall: () => new Promise(res => res({ success: true })),
    apiCallPendingMsg: "잠시만 기다려주세요",
    apiCallErrMsg: "사용할 수 없는 값입니다"
  },
  match: {
    matchVal: "",
    matchValErrMsg: "값이 일치하지 않습니다"
  }
  }
  
  const useValidateInput = (
  initalval, 
  emptyValidation= VALIDATAION_INITIAL_VALES[empty],
  ...(중략) 
  )
```

이런식은 어떤가요?! 그런데 파라미터 순서를 기억하기 힘드니 
```js
const useValidateInput = ( initValue, option = VALIDATAION_INITIAL_VALES  ) = {
const { empty, reg, api, match } = option 
}
```
추천드려요 ㅎㅎ